ui/src/definitely-typed/filters/definitelyTypedFilters.tsx

```diff
@@ -2,80 +2,174 @@ import React, { useEffect, useState } from 'react';
 import './definitelyTypedFilters.scss';
 import '../input/input.scss';
 import { useDebounce } from '../../core/useDebounce/useDebounce';
+import { Option, Select } from '../../style/select/select.styled';
+import { DefinitelyTypedTypeRun } from "../interfaces/definitelyTypedTypeRun.interface";
+import { PresetFilter } from "./errorPresetFilters";
 
 export interface DefinitelyTypedRunInfo {
-    success: number;
-    warning: number;
-    error: number;
-    total: number;
+  success: number;
+  warning: number;
+  error: number;
+  total: number;
 }
 
 export interface DefinitelyTypedFiltersProps {
-    runInfo: DefinitelyTypedRunInfo;
-    filter(options: DefinitelyTypedFiltersOptions): void;
+  runInfo: DefinitelyTypedRunInfo;
+  data: DefinitelyTypedTypeRun[];
+  filter(options: DefinitelyTypedFiltersOptions): void;
 }
 
 export interface DefinitelyTypedFiltersOptions {
-    filterIn: RegExp | null;
-    filterOut: RegExp | null;
-    isShowingSuccesses: boolean;
-    isShowingWarnings: boolean;
-    isShowingErrors: boolean;
+  filterIn: RegExp | null;
+  filterOut: RegExp | null;
+  isShowingSuccesses: boolean;
+  isShowingWarnings: boolean;
+  isShowingErrors: boolean;
 }
 
-export function DefinitelyTypedFilters(props: DefinitelyTypedFiltersProps): JSX.Element {
-    const [filterOut, setFilterOut] = useState('');
-    const [filterIn, setFilterIn] = useState('');
-    const [isShowing, setIsShowing] = useState('' as keyof DefinitelyTypedRunInfo);
-    const debouncedFilterOut = useDebounce(filterOut, 500);
-    const debouncedFilterIn = useDebounce(filterIn, 500);
-
-    useEffect(() => {
-        props.filter(buildOptions());
-    }, [debouncedFilterOut, debouncedFilterIn, isShowing]);
-
-    function buildOptions(): DefinitelyTypedFiltersOptions {
-        return {
-            filterIn: filterIn ? new RegExp(filterIn, 'i') : null,
-            filterOut: filterOut ? new RegExp(filterOut, 'i') : null,
-            isShowingErrors: isShowing === 'error',
-            isShowingWarnings: isShowing === 'warning',
-            isShowingSuccesses: isShowing === 'success'
-        };
-    }
+export function DefinitelyTypedFilters(
+  props: DefinitelyTypedFiltersProps
+): JSX.Element {
+  const [filterOut, setFilterOut] = useState('');
+  const [filterIn, setFilterIn] = useState('');
+  const [errorPresetFilterId, setErrorPresetFilterId] = useState(0);
```

Comment:
I think it's going to be difficult in the future to understand. Shall we convert to an enum at least? None = 0, Custom = 1 and we don't care about the rest

Reply:
created an enum with -1 and -2 as values so we don't have to do -3 in the calculations